src/tuple_gen.cpp

```diff
@@ -177,23 +181,38 @@ void CIpInfoBase:: detach_pool_and_client_node(void *ip_pool){
         }
     }
 }
-/* base on thread_id and client_index 
-  client index would be the same for all thread 
-*/
+
+std::random_device rand_dev;
+std::mt19937 rand_gen{rand_dev()};
+std::uniform_int_distribution<uint16_t> uniform_port_dist(MIN_PORT, MAX_PORT);
+
+/**
+ * Based on thread_id and client_index, or unfiform distribution. 
+ * Client index would be the same for all threads. 
+ *
+ * @param rand_client_port If true, use random port with uniform distribution instead of thread_id and client_id.
+ */
 static uint16_t generate_rand_sport(uint32_t client_index,
                                     uint16_t threadid,
                                     uint16_t rss_thread_id,
                                     uint16_t rss_thread_max,
                                     uint8_t  reta_mask,
-                                    bool     rss_astf_mode){
+                                    bool     rss_astf_mode,
+                                    bool     rand_client_port){
+    uint16_t port;
 
-    uint32_t rand  = ((214013*(client_index + 11213*threadid) + 2531011));
-    uint16_t rand16 = ((rand)&0xffff);
-    uint16_t port = MIN_PORT+(rand16%(MAX_PORT-MIN_PORT))+1;
+    if (rand_client_port) {
+        port = uniform_port_dist(rand_gen);
+    } else {
+        uint32_t rand  = ((214013*(client_index + 11213*threadid) + 2531011));
+        uint16_t rand16 = ((rand)&0xffff);
+        port = MIN_PORT+(rand16%(MAX_PORT-MIN_PORT))+1;
```

Comment:
Now that we're randomising the value in the othere branch it would be great to name these variables for what they are
```suggestion
        uint32_t const hash32 = 214013 * (client_index + 11213 * threadid) + 2531011;
        uint16_t const hash16 = hash32 & 0xffff;
        port = MIN_PORT + (hash16 % (MAX_PORT - MIN_PORT)) + 1;
```

Reply:
Done